source/frontend/src/features/documents/list/DocumentActivityListContainer.tsx

```diff
@@ -0,0 +1,129 @@
+import { noop } from 'lodash';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+import usePathGenerator from '@/features/mapSideBar/shared/sidebarPathGenerator';
+import { useManagementActivityRepository } from '@/hooks/repositories/useManagementActivityRepository';
+import { usePimsPropertyRepository } from '@/hooks/repositories/usePimsPropertyRepository';
+import useIsMounted from '@/hooks/util/useIsMounted';
+import { ApiGen_CodeTypes_DocumentRelationType } from '@/models/api/generated/ApiGen_CodeTypes_DocumentRelationType';
+import { ApiGen_Concepts_DocumentRelationship } from '@/models/api/generated/ApiGen_Concepts_DocumentRelationship';
+import { ApiGen_Concepts_PropertyActivity } from '@/models/api/generated/ApiGen_Concepts_PropertyActivity';
+import { exists, getApiPropertyName, relationshipTypeToPathName } from '@/utils';
+
+import { DocumentRow } from '../ComposedDocument';
+import { useDocumentRelationshipProvider } from '../hooks/useDocumentRelationshipProvider';
+import { IDocumentListContainerProps } from './DocumentListContainer';
+import DocumentListView from './DocumentListView';
+import { DocRelation } from './models';
+
+const DocumentActivityListContainer: React.FunctionComponent<
+  IDocumentListContainerProps
+> = props => {
+  const pathGenerator = usePathGenerator();
+
+  const isMounted = useIsMounted();
+
+  const [documentResults, setDocumentResults] = useState<DocumentRow[]>([]);
+
+  const [activity, setActivity] = useState<ApiGen_Concepts_PropertyActivity>(null);
+
+  const {
+    getPropertyManagementActivity: { execute: getManagementActivity, loading: getActivityLoading },
+  } = useManagementActivityRepository();
+
+  const {
+    getAllPropertiesById: { execute: getProperties, loading: getPropertiesLoading },
+  } = usePimsPropertyRepository();
+
+  const { retrieveDocumentRelationship, retrieveDocumentRelationshipLoading } =
+    useDocumentRelationshipProvider();
+
+  const retrieveActivities = useCallback(async () => {
+    const result = await getManagementActivity(Number(props.parentId));
+    if (exists(result) && isMounted()) {
+      const propertyResult = await getProperties(
+        result.activityProperties.map(ap => ap.propertyId),
+      );
+      result.activityProperties.forEach(ap => {
```

Comment:
techically this could be gated by isMounted too, but we haven't been consistent on this.

Reply:
(the main thing you want to prevent is state updates after awaits)